sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/BeamZetaSqlCalcRel.java

```diff
@@ -201,23 +199,22 @@ public void setup() {
             ZetaSqlBeamTranslationUtils.toZetaSqlType(inputSchema.getField(i).getType()));
       }
 
-      exp = new PreparedExpression(sql);
-      exp.prepare(options);
+      PreparedExpression expression = new PreparedExpression(sql);
+      exp = expression;
```

Comment:
I'm not really a fan of this. I believe you can move `self.exp = new PreparedExpression(sql);` to the constructor so `exp` doesn't need to be `@Nullable`? (You'll also need to add the line to `teardown` as a `PreparedExpression` can't be reused.) Then you should be able to revert the rest of the changes to this method. If that doesn't work, I'd prefer nullness checks in this method for consistency.

Reply:
Done.